### PR TITLE
Add hugepages e2e tests to NodeConformance

### PR DESCRIPTION
**Pull request by bart0sh**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

It promotes hugepages tests to node conformance suite. This is a requirement for [multi-size hugepages promotion to GA](https://github.com/kubernetes/enhancements/pull/2534).

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

**Comment by pacoxu** [comment]
For conformance testing changes, you shoudl run `./hack/verify-conformance-yaml.sh` to update the conformance yaml.

**Comment by bart0sh** [comment]
@pacoxu thank you for the review. I've updated the PR accordingly.

**Comment by dims** [comment]
/assign @dchen1107 @derekwaynecarr 

Derek, Dawn, is this something you want to add to the conformance suite? (not just `NodeConformance`)?

**Comment by bart0sh** [comment]
@dims Adding this to Conformance would require moving test code to the test/e2e/common/node/, updating the code as the APIs are different and updating the job description in test-infra repo.

I thought that if this test was initially put into test/e2e_node/ it means that promotion to conformance was not considered and it should be promoted to NodeConformance only.

File: test/e2e_node/hugepages_test.go [hunk 1]
```diff
@@ -192,7 +192,12 @@ func getHugepagesTestPod(f *framework.Framework, limits v1.ResourceList, mounts
 var _ = SIGDescribe("HugePages [Serial] [Feature:HugePages][NodeSpecialFeature:HugePages]", func() {
 	f := framework.NewDefaultFramework("hugepages-test")
 
-	ginkgo.It("should remove resources for huge page sizes no longer supported", func() {
+	/*
+	   Release : v1.22
+	   Testname: Removal of hugepages resources
+	   Description: Kubelet should remove hugepages resources of deallocated hugepages.
+	*/
+	framework.ConformanceIt("should remove resources for huge page sizes no longer supported [NodeConformance]", func() {
```
Comment (ehashman) [inline test/e2e_node/hugepages_test.go:200]: Is this a [LinuxOnly] feature?

**Review by ehashman — COMMENTED** [review]
/triage accepted
/priority important-soon

**Comment by ehashman** [comment]
/lgtm cancel
given the questions above

File: test/e2e_node/hugepages_test.go (hunk 1 above)
Comment (bart0sh) [reply test/e2e_node/hugepages_test.go:200]: Thank you for the review.
It is. Updated the PR.

**Comment by SergeyKanzhelev** [comment]
/hold
let's understand why this test is not being picked up by conformance. I wonder if this test can even be a conformance as it likely uses special environment? @bart0sh can you help drive this understanding?

**Comment by bart0sh** [comment]
@SergeyKanzhelev yes, this test requires preallocation of 1Gb HugePages. It's configured in [this file](https://github.com/kubernetes/test-infra/blob/master/jobs/e2e_node/ubuntu-hugepages-1G-allocation.yaml) and referred in the [hugepages job](https://github.com/kubernetes/test-infra/blob/master/jobs/e2e_node/image-config-serial-hugepages.yaml#L8)
@cynepco3hahue am I correct with this?

**Comment by SergeyKanzhelev** [comment]
> @SergeyKanzhelev no idea why it wasn't picked up. where can I see this?

all conformance tests are listed in https://github.com/kubernetes/kubernetes/blob/master/test/conformance/testdata/conformance.yaml This PR doesn't include the change of this file. Thus is wasn't picked up. Likely because it is in `test/e2e_node`, not in `test/e2e/node`.

So if part of feature GA we need to add conformance test, I wonder if it can be run there (I didn't look what this test does).

**Comment by SergeyKanzhelev** [comment]
there is a #k8s-conformance channel. Or maybe @hh can advice.

**Comment by cynepco3hahue** [comment]
> @cynepco3hahue am I correct with this?

Yes, it correct, so it the Linux only.

**Comment by bart0sh** [comment]
@SergeyKanzhelev Just to make sure we're on the same page here.
This PR adds hugepages tests to NodeConformance, not to Conformance. Adding them to Conformance would require changing test code a lot. See more details [here](https://github.com/kubernetes/kubernetes/pull/102774#issuecomment-860742706)

**Comment by bart0sh** [comment]
@SergeyKanzhelev @ehashman I've asked about promoting this test to Conformance on the #k8s-conformance. Here is their answer: https://kubernetes.slack.com/archives/C78F00H99/p1624439068035700?thread_ts=1624436955.035500&cid=C78F00H99
```
bentheelder  4 hours ago
I don't think we can have a conformance test that requires special configuration of the cluster nodes ...
```

**Comment by bart0sh** [comment]
Next question: can we have a test that requires special configuration in the NodeConformance?

**Comment by bart0sh** [comment]
@ehashman @SergeyKanzhelev any suggestions how to proceed further with this PR? Promoting this test to Conformance is not possible. NodeConformance doesn't seem to be a good place for it neither. Should we close it?

**Comment by ehashman** [comment]
I think we should close it, @SergeyKanzhelev to confirm.

**Comment by SergeyKanzhelev** [comment]
/close

agree, thank you!

**Pull request closed without merging**